content/html-css/blocks/what-is-css/index.md

```diff
@@ -0,0 +1,118 @@
+}
+```
+
+Selectors look like this: `p { background: red }`. The selector is the `p` in this case. This p selects all elements of the DOM tree called `p` and sets a rule about how to show them. Selectors always come first in a ruleset.
+
+A property is a quality, a characteristic. A value is the amount.
+
+For a person, you might say their 'age' property has the value of 40. For an element, you might set the `width` property to the value of `100%`.
+
+</details>
+
+<details>
+<summary>How do you know what properties and values there are?</summary>
+
+You learn them by using them. You look it up here: https://developer.mozilla.org/en-US/docs/Web/CSS/Reference. And the other thing you can do is look at the autocomplete in Devtools.
+
+</details>
+
+<details>
+<summary>Why is CSS a 'cascading' style sheet? What does cascade mean?</summary>
+
+<iframe src="https://giphy.com/embed/zFSGphnnT9opa" width="480" height="480" frameBorder="0" class="giphy-embed" allowFullScreen></iframe><p><a href="https://giphy.com/gifs/zFSGphnnT9opa">via GIPHY</a></p>
+
+A cascade is a stream or sequence of a things where each thing affects the next. You might imagine pouring champagne into a tower of glasses. The champagne flows into the containers; it cascades. In CSS we’re actually pouring from many bottles into one tower.
+
+So in CSS 'the cascade' means the order or sequence of the rules as they flow into the DOM tree according to CSS precedence.
+
+</details>
+
+<details>
+<summary>What does precedence mean?</summary>
+
+It means hierarchy. One hierarchy in CSS is inheritance: children inherit from parents. If a parent node has red text, all its children will inherit that red text color. But when another rule comes after, and overrules, it takes precedence. Everything in CSS is determined by order.
+
+Order is so important in CSS because CSS is a declarative programming language that programs the layout of boxes.
+
+CSS renders the nodes in our DOM tree as nested boxes, and we program their appearance using fonts, colours, and spacing.
+
+</details>
+
+<details>
+<summary>What does declarative mean?</summary>
+
+It means, broadly speaking, that the entire code is processed and then applied, from top to bottom in one go. You don’t write for loops or if/else statements; there are no functions as such; you just _declare_ the outcome.
+
+In CSS you do the 'what', and the browser does the 'how'.
+
+</details>
+
+<details>
+<summary>What is specificity?</summary>
+
+Specificity is a [weighting system for CSS rules](https://specifishity.com/). The more specific your selector, the more power it has. For example: IDs can only ever mean _one_ node. IDs are unique, so they are _more specific_ than classes, which could mean many nodes.
+
+Classes added to html element are _more specific_ than just a plain element selector, as a class of `p` is _more specific_ than just all `p`. If you get two elements with the same specificity then the one that comes last in the order wins.
```

Comment:
I'm struggling to understand this point?

I think it's possibly the phrase "class of `p`": initially I thought you were using an example something like this: `<p class="p">...</p>`, which is quite confusing.

But re-reading it I think you're using "class" in the abstract sense here: all `p` in a (potential) class? Assuming that's true, the word "class" is quite overloaded here: it refers to the literal keyword and the abstract concept.

What about rewording to something like:

```suggestion
Classes added to html elements are _more specific_ than just a plain element selector, as a `p` with a class is _more specific_ than just all possible `p`s. If you get two elements with the same specificity then the one that comes last in the order wins.
```